integrations/autotools/building_with_autotools.rst

```diff
@@ -0,0 +1,318 @@
+.. _building_with_autotools:
+
+|gnu_logo|
+
+Building with Autotools
+=======================
+
+If you are using **configure**/**make** you can use **AutoToolsBuildEnvironment** helper.
+This helper sets ``LIBS``, ``LDFLAGS``, ``CFLAGS``, ``CXXFLAGS`` and ``CPPFLAGS`` environment variables based on your requirements.
+
+.. code-block:: python
+   :emphasize-lines: 13, 14, 15
+
+   from conans import ConanFile, AutoToolsBuildEnvironment
+
+   class ExampleConan(ConanFile):
+      settings = "os", "compiler", "build_type", "arch"
+      requires = "poco/1.9.4"
+      default_options = {"poco:shared": True, "openssl:shared": True}
+
+      def imports(self):
+         self.copy("*.dll", dst="bin", src="bin")
+         self.copy("*.dylib*", dst="bin", src="lib")
+
+      def build(self):
+         autotools = AutoToolsBuildEnvironment(self)
+         autotools.configure()
+         autotools.make()
+
+It also works using the (:ref:)`environment_append <tools_environment_append>` context manager applied to your **configure and make** commands,
+calling `configure` and `make` manually:
+
+.. code-block:: python
+
+    from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+    class ExampleConan(ConanFile):
+        ...
+
+        def build(self):
+            env_build = AutoToolsBuildEnvironment(self)
+            with tools.environment_append(env_build.vars):
+                self.run("./configure")
+                self.run("make")
+
+You can change some variables like ``fpic``, ``libs``, ``include_paths`` and ``defines`` before accessing the ``vars`` to override
+an automatic value or add new values:
+
+.. code-block:: python
+   :emphasize-lines: 8, 9, 10
+
+    from conans import ConanFile, AutoToolsBuildEnvironment
+
+    class ExampleConan(ConanFile):
+        ...
+
+        def build(self):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.fpic = True
+            env_build.libs.append("pthread")
+            env_build.defines.append("NEW_DEFINE=23")
+            env_build.configure()
+            env_build.make()
+
+You can use it also with ``MSYS2``/``MinGW`` subsystems installed by setting the `win_bash` parameter
+in the constructor. It will run the the ``configure`` and ``make`` commands inside a ``bash`` that
+has to be in the path or declared in ``CONAN_BASH_PATH``:
+
+
+.. code-block:: python
+   :emphasize-lines: 12
+
+   from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+   class ExampleConan(ConanFile):
+      settings = "os", "compiler", "build_type", "arch"
+
+      def imports(self):
+        self.copy("*.dll", dst="bin", src="bin")
+        self.copy("*.dylib*", dst="bin", src="lib")
+
+      def build(self):
+         env_build = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+         env_build.configure()
+         env_build.make()
+
+Constructor
```

Comment:
This is reference-like docs, it shouldn't go here, but redirect to the ``reference`` section.